### PR TITLE
PT-1924: Add rounding policy

### PR DESCRIPTION
**Pull request by krankenbro**

### CODE REVIEW

**Comment by Vectorfield4** [comment]
Write Unit-tests for `DefaultMoneyRoundingPolicy`

File: src/VirtoCommerce.CoreModule.Core/Currency/Currency.cs [hunk 1]
```diff
@@ -115,6 +115,10 @@ public string CultureName
         /// </summary>
         public string CustomFormatting { get; set; }
 
+        public IMoneyRoundingPolicy RoundingPolicy { get; set; }
```
Comment (tatarincev) [inline src/VirtoCommerce.CoreModule.Core/Currency/Currency.cs:118]: Please add [JsonIgnore] attribute

File: src/VirtoCommerce.CoreModule.Core/Currency/RoundingType.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+namespace VirtoCommerce.CoreModule.Core.Currency
+{
+    public enum RoundingType
```
Comment (tatarincev) [inline src/VirtoCommerce.CoreModule.Core/Currency/RoundingType.cs:3]: add comment please //Cash rounding (details: https://en.wikipedia.org/wiki/Cash_rounding)

File: src/VirtoCommerce.CoreModule.Data/Currency/CurrencyService.cs [hunk 3]
```diff
@@ -17,12 +17,14 @@ public class CurrencyService : ICurrencyService
...
         private readonly IPlatformMemoryCache _platformMemoryCache;
+        private readonly IMoneyRoundingPolicy _moneyRoundingPolicy;
 
-        public CurrencyService(Func<ICoreRepository> repositoryFactory, IEventPublisher eventPublisher, IPlatformMemoryCache platformMemoryCache)
+        public CurrencyService(Func<ICoreRepository> repositoryFactory, IEventPublisher eventPublisher, IPlatformMemoryCache platformMemoryCache, IMoneyRoundingPolicy moneyRoundingPolicy)
         {
             _repositoryFactory = repositoryFactory;
             _eventPublisher = eventPublisher;
             _platformMemoryCache = platformMemoryCache;
+            _moneyRoundingPolicy = moneyRoundingPolicy;
```
Comment (tatarincev) [inline src/VirtoCommerce.CoreModule.Data/Currency/CurrencyService.cs:27]: Missed  `_moneyRoundingPolicy` initialization  to returned currencies from the service

**Review by tatarincev — CHANGES_REQUESTED** [review]

**Review by Vectorfield4 — APPROVED** [review]
You can merge this PR after fixing code smells.

**Pull request merged**